lib/job-iteration/iteration.rb

```diff
@@ -291,13 +270,22 @@ def instrumentation_tags
     end
 
     def job_should_exit?
-      if job_iteration_max_job_runtime && start_time && (Time.now.utc - start_time) > job_iteration_max_job_runtime
-        return true
-      end
+      max_job_runtime = job_iteration_max_job_runtime
+      return true if max_job_runtime && start_time && (Time.now.utc - start_time) > max_job_runtime
 
       JobIteration.interruption_adapter.call || (defined?(super) && super)
     end
 
+    def job_iteration_max_job_runtime
+      global_max = JobIteration.max_job_runtime
+      class_max = self.class.job_iteration_max_job_runtime
+
+      return global_max unless class_max
+      return class_max unless global_max
+
+      [global_max, class_max].min
```

Comment:
@Mangara I benchmarked this compared to `compact`, `compact!`, and nested conditionals. This is faster than both `compact` versions, and nearly identical to nested conditionals, so I changed it from our `compact!` version and went with this as I figure it's pretty clear, faster, and avoids allocating arrays entirely.

Reply:
Yep, that makes sense to me! I don't think performance is super critical here, as most iterations will contain some external calls (DB queries mostly) that will dominate the running time. But this is still very readable, so if we can get both performance and readability I won't say no 😄